drivers/i2c/i2c_gecko.c

```diff
@@ -121,6 +126,8 @@ static int i2c_gecko_transfer(const struct device *dev, struct i2c_msg *msgs, ui
 			seq.flags |= I2C_FLAG_10BIT_ADDR;
 		}
 
+		k_sem_take(&data->bus_mutex, K_FOREVER);
```

Comment:
The lock should wrap the entirety of transfer, what happens when two threads call transfer simultaneously otherwise?

Reply:
you are right : updated!